### PR TITLE
upgrade to akka 1.3.0

### PR DESCRIPTION
**Pull request by davesearle**

### CODE REVIEW

**Comment by Horusiath** [comment]
Thanks, this is valuable help for the rest of the team! I think, we may need to update CI here to make build steps pass on .NET Standard. /cc @heynickc @Aaronontheweb

File: src/Akka.Persistence.SqlServer.Tests/app.config [hunk 1]
```diff
@@ -21,6 +21,10 @@
         <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-1.2.1.0" newVersion="1.2.1.0" />
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.SqlServer.Tests/app.config]: This file could be removed

File: src/Akka.Persistence.SqlServer/packages.config [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.SqlServer/packages.config]: This file could be removed

File: src/Akka.Persistence.SqlServer/Akka.Persistence.SqlServer.csproj [hunk 3]
```diff
@@ -1,109 +1,44 @@
...
+      <GenerateDocumentationFile>true</GenerateDocumentationFile>
+      <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.6' ">1.6.1</NetStandardImplicitPackageVersion>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Akka" Version="1.3.0" />
+      <PackageReference Include="Akka.Persistence" Version="1.3.0" />
+      <PackageReference Include="Akka.Persistence.Sql.Common" Version="1.3.0" />
+      <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
+      <PackageReference Include="protobuf" Version="2.6.1.1" />
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.SqlServer/Akka.Persistence.SqlServer.csproj]: Why we need a protobuf here?

File: src/Akka.Persistence.SqlServer.Tests/Akka.Persistence.SqlServer.Tests.csproj [hunk 4]
```diff
@@ -1,203 +1,46 @@
...
-      <Project>{bac85686-afc4-413e-98dc-5ed8f468bc63}</Project>
-      <Name>Akka.Persistence.SqlServer</Name>
-    </ProjectReference>
-  </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+<Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
-      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+      <AssemblyTitle>Akka.Persistence.SqlServer.Tests</AssemblyTitle>
```
Comment (alexvaluyskiy) [inline src/Akka.Persistence.SqlServer.Tests/Akka.Persistence.SqlServer.Tests.csproj]: Is it not necessary to put the description/tags/etc for tests projects

File: src/Akka.Persistence.SqlServer/Akka.Persistence.SqlServer.csproj (hunk 3 above)
Comment (davesearle) [inline src/Akka.Persistence.SqlServer/Akka.Persistence.SqlServer.csproj]: Oversight when looking at previous referenced packages - have removed it, along with Newtonsoft

File: src/Akka.Persistence.SqlServer.Tests/Akka.Persistence.SqlServer.Tests.csproj (hunk 4 above)
Comment (davesearle) [inline src/Akka.Persistence.SqlServer.Tests/Akka.Persistence.SqlServer.Tests.csproj]: Removed

File: src/Akka.Persistence.SqlServer/packages.config (hunk 2 above)
Comment (davesearle) [inline src/Akka.Persistence.SqlServer/packages.config]: Removed

File: src/Akka.Persistence.SqlServer.Tests/app.config (hunk 1 above)
Comment (davesearle) [inline src/Akka.Persistence.SqlServer.Tests/app.config]: Removed

**Comment by Aaronontheweb** [comment]
Looking at the build system...

**Comment by Aaronontheweb** [comment]
Update on this - got most everything converted over to .NET Standard / .NET Core but I'm going to need to tweak the test suite a bit to make it's use of the `ConfigurationManager` .NET-Core friendly using the new libraries. Hoping to have the whole Docker + integration test suite functioning tomorrow.

**Comment by Aaronontheweb** [comment]
Have integration tests successfully connecting to Docker again, after a tremendous amount of effort lol...

```
[ERROR][8/18/2017 10:58:40 PM][Thread 0016][[akka://SqlServerJournalSpec/system/akka.persistence.journal.sql-server#248849433]] Failure during [akka://SqlServerJournalSpec/system/akka.persistence.journal.sql-server#248849433] initialization.
Cause: System.NullReferenceException: Object reference not set to an instance of an object.
   at Akka.Persistence.Sql.Common.Journal.SqlJournal.GetConnectionString()
   at Akka.Persistence.Sql.Common.Journal.SqlJournal.<Initialize>d__36.MoveNext()
```

Looks to me like there's an issue inside Akka.Persistence.Sql.Common here with being able to pull in the correct connection string... Going to drill in a bit deeper, but it looks like this issue isn't necessarily specific to the Akka.Persistence.Sql plugin.

**Comment by Aaronontheweb** [comment]
I take that back... it's an issue with `connection-string-name` not working anymore without the `ConfigurationManager`. If you're using `Microsoft.Extensions.Configuration` you won't be able to pull in configuration strings by name. They'll need to be explicitly included in the HOCON.

**Comment by Danthar** [comment]
This PR will go a long way to bringing this plugin up to date. 
However there is still work to be done: https://github.com/akkadotnet/Akka.Persistence.SqlServer/issues/74

**Comment by Horusiath** [comment]
Why all tests failed? Wrong connection string?

**Comment by Aaronontheweb** [comment]
@Horusiath yeah the Journal isn't pulling it in correctly right now and the snapshotstore is connecting to the wrong database

**Comment by Horusiath** [comment]
Guys. We need this PR.

**Comment by heynickc** [comment]
@Horusiath I'm working on this now that we have 1.3.1 released

**Comment by heynickc** [comment]
ok, CI should work with my latest changes pushed to your branch @davesearle - I'll keep an eye on it to make sure

**Comment by heynickc** [comment]
I'm working on moving this PR up to 1.3.1 at the moment so please don't merge yet

**Comment by heynickc** [comment]
@Horusiath or anyone more familiar with `Persistence.Query`, I updated this PR to target Akka v1.3.1, however there are 2 specs failing referring to Akka.Streams that I am going to need help with debugging:
```
Starting test execution, please wait...
[xUnit.net 00:00:00.6412957]   Discovering: Akka.Persistence.SqlServer.Tests
[xUnit.net 00:00:00.7494999]   Discovered:  Akka.Persistence.SqlServer.Tests
[xUnit.net 00:00:00.7542315]   Starting:    Akka.Persistence.SqlServer.Tests
[xUnit.net 00:00:25.2509766]     Akka.Persistence.SqlServer.Tests.Query.SqlServerEventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_new_events [F
AIL]
[xUnit.net 00:00:25.2520068]       Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnNext`1[Akka.Persistence.Query.EventEnvelope], but
 received {TestSubscriber.OnError(One or more errors occurred. (Unable to cast object of type 'System.Int64' to type 'System.Int32'.))} (type Akka.Streams.Te
stKit.TestSubscriber+OnError) instead  from [akka://SqlServerEventsByTagSpec/user/StreamSupervisor-10/Flow-0-0-actorPublisherSource-EventsByTag-black-actorPu
blisherSource#1550733812]
[xUnit.net 00:00:25.2521983]       Expected: True
[xUnit.net 00:00:25.2522406]       Actual:   False
[xUnit.net 00:00:25.2534652]       Stack Trace:
[xUnit.net 00:00:25.2545625]            at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args)
[xUnit.net 00:00:25.2546565]            at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean sh
ouldLog)
[xUnit.net 00:00:25.2546967]            at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`2 assert, String hint)
[xUnit.net 00:00:25.2547455]            at Akka.TestKit.TestKitBase.ExpectMsg[T](Predicate`1 isMessage, Nullable`1 timeout, String hint)
[xUnit.net 00:00:25.2547783]            at Akka.Streams.TestKit.TestSubscriber.ManualProbe`1.ExpectNext[TOther](Predicate`1 predicate)
[xUnit.net 00:00:25.2548079]            at Akka.Persistence.TCK.Query.EventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_new_events()
[xUnit.net 00:00:25.2556645]       Output:
[xUnit.net 00:00:25.2557498]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0014][ActorSystem(SqlServerEventsByTagSpec)] System shutdown initiated
[xUnit.net 00:00:25.2557812]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0012][EventStream] Shutting down: StandardOutLogger started
[xUnit.net 00:00:25.2558042]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0012][EventStream] All default loggers stopped
[xUnit.net 00:00:25.3731353]     Akka.Persistence.SqlServer.Tests.Query.SqlServerEventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_events_from_o
ffset_exclusive [FAIL]
[xUnit.net 00:00:25.3733075]       Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnNext`1[Akka.Persistence.Query.EventEnvelope], but
 received {TestSubscriber.OnError(One or more errors occurred. (Unable to cast object of type 'System.Int64' to type 'System.Int32'.))} (type Akka.Streams.Te
stKit.TestSubscriber+OnError) instead  from [akka://SqlServerEventsByTagSpec/user/StreamSupervisor-11/Flow-0-0-actorPublisherSource-EventsByTag-green-actorPu
blisherSource#1117816760]
[xUnit.net 00:00:25.3734034]       Expected: True
[xUnit.net 00:00:25.3734293]       Actual:   False
[xUnit.net 00:00:25.3734603]       Stack Trace:
[xUnit.net 00:00:25.3735055]            at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args)
[xUnit.net 00:00:25.3735664]            at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean sh
ouldLog)
[xUnit.net 00:00:25.3736156]            at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`2 assert, String hint)
[xUnit.net 00:00:25.3736627]            at Akka.TestKit.TestKitBase.ExpectMsg[T](Predicate`1 isMessage, Nullable`1 timeout, String hint)
[xUnit.net 00:00:25.3736977]            at Akka.Streams.TestKit.TestSubscriber.ManualProbe`1.ExpectNext[TOther](Predicate`1 predicate)
[xUnit.net 00:00:25.3737400]            at Akka.Persistence.TCK.Query.EventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_events_from_offset_exclu
sive()
[xUnit.net 00:00:25.3737812]       Output:
[xUnit.net 00:00:25.3738151]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0014][ActorSystem(SqlServerEventsByTagSpec)] System shutdown initiated
[xUnit.net 00:00:25.3738468]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0040][EventStream] Shutting down: StandardOutLogger started
[xUnit.net 00:00:25.3738815]         [DEBUG][9/6/2017 9:21:45 PM][Thread 0040][EventStream] All default loggers stopped
[xUnit.net 00:00:25.3761741]   Finished:    Akka.Persistence.SqlServer.Tests
Failed   Akka.Persistence.SqlServer.Tests.Query.SqlServerEventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_new_events
Error Message:
 Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnNext`1[Akka.Persistence.Query.EventEnvelope], but received {TestSubscriber.OnError(
One or more errors occurred. (Unable to cast object of type 'System.Int64' to type 'System.Int32'.))} (type Akka.Streams.TestKit.TestSubscriber+OnError) inst
ead  from [akka://SqlServerEventsByTagSpec/user/StreamSupervisor-10/Flow-0-0-actorPublisherSource-EventsByTag-black-actorPublisherSource#1550733812]
Expected: True
Actual:   False
Stack Trace:
   at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args)
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean shouldLog)
   at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`2 assert, String hint)
   at Akka.TestKit.TestKitBase.ExpectMsg[T](Predicate`1 isMessage, Nullable`1 timeout, String hint)
   at Akka.Streams.TestKit.TestSubscriber.ManualProbe`1.ExpectNext[TOther](Predicate`1 predicate)
   at Akka.Persistence.TCK.Query.EventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_new_events()Standard Output Messages:
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0014][ActorSystem(SqlServerEventsByTagSpec)] System shutdown initiated
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0012][EventStream] Shutting down: StandardOutLogger started
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0012][EventStream] All default loggers stopped

Failed   Akka.Persistence.SqlServer.Tests.Query.SqlServerEventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_events_from_offset_exclusive
Error Message:
 Failed: Expected a message of type Akka.Streams.TestKit.TestSubscriber+OnNext`1[Akka.Persistence.Query.EventEnvelope], but received {TestSubscriber.OnError(
One or more errors occurred. (Unable to cast object of type 'System.Int64' to type 'System.Int32'.))} (type Akka.Streams.TestKit.TestSubscriber+OnError) inst
ead  from [akka://SqlServerEventsByTagSpec/user/StreamSupervisor-11/Flow-0-0-actorPublisherSource-EventsByTag-green-actorPublisherSource#1117816760]
Expected: True
Actual:   False
Stack Trace:
   at Akka.TestKit.Xunit2.XunitAssertions.Fail(String format, Object[] args)
   at Akka.TestKit.TestKitBase.InternalExpectMsgEnvelope[T](Nullable`1 timeout, Action`2 assert, String hint, Boolean shouldLog)
   at Akka.TestKit.TestKitBase.InternalExpectMsg[T](Nullable`1 timeout, Action`2 assert, String hint)
   at Akka.TestKit.TestKitBase.ExpectMsg[T](Predicate`1 isMessage, Nullable`1 timeout, String hint)
   at Akka.Streams.TestKit.TestSubscriber.ManualProbe`1.ExpectNext[TOther](Predicate`1 predicate)
   at Akka.Persistence.TCK.Query.EventsByTagSpec.ReadJournal_live_query_EventsByTag_should_find_events_from_offset_exclusive()Standard Output Messages:
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0014][ActorSystem(SqlServerEventsByTagSpec)] System shutdown initiated
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0040][EventStream] Shutting down: StandardOutLogger started
 [DEBUG][9/6/2017 9:21:45 PM][Thread 0040][EventStream] All default loggers stopped


Total tests: 40. Passed: 38. Failed: 2. Skipped: 0.
Test Run Failed.
Test execution time: 26.0852 Seconds
```

The biggest change here is the `SerializerId` column, which is of type `INTEGER` so that might be some source of issue.

**Comment by heynickc** [comment]
This is ready to merge once CI passes

**Pull request merged**